File: bae-android/app/src/main/java/fm/bae/app/ui/RepeatModeCycle.kt
package fm.bae.app.ui

import uniffi.bae_bridge.BridgeRepeatMode

/** The next mode in a repeat button's cycle: OFF → CONTEXT → TRACK → OFF.
 *  UI-owned: core only accepts absolute setRepeatMode values; the caller
 *  computes the target from the mode it renders. */
fun BridgeRepeatMode.next(): BridgeRepeatMode =
    when (this) {
        BridgeRepeatMode.OFF -> BridgeRepeatMode.CONTEXT
        BridgeRepeatMode.CONTEXT -> BridgeRepeatMode.TRACK
        BridgeRepeatMode.TRACK -> BridgeRepeatMode.OFF
    }

File: BaeKit/Sources/BaeKit/Services/Store/RepeatModeCycle.swift
extension BridgeRepeatMode {
    /// The next mode in a repeat button's cycle: off → context → track → off.
    /// UI-owned: core only accepts absolute `setRepeatMode` values; the caller
    /// computes the target from the mode it renders.
    public var next: BridgeRepeatMode {
        switch self {
        case .off: .context
        case .context: .track
        case .track: .off
        }
    }
}
